src/app/modules/project/highlights-modal/highlights-modal.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+import { Highlight } from '../../../models/domain/highlight';
+import { HighlightService } from 'src/app/services/highlight.service';
+import { AlertService } from 'src/app/services/alert.service';
+import { AlertType } from 'src/app/models/internal/alert-type';
+import { AlertConfig } from 'src/app/models/internal/alert-config';
+import { faTrash } from '@fortawesome/free-solid-svg-icons';
+
+@Component({
+  selector: 'app-highlights-modal',
+  templateUrl: './highlights-modal.component.html',
+  styleUrls: ['./highlights-modal.component.scss']
+})
+export class HighlightsModalComponent {
+  @Input() highlights: Highlight[];
+  @Output() selectHighlightToEdit = new EventEmitter();
+  @Output() selectAddHighlight = new EventEmitter();
+
+  public faTrash = faTrash;
+
+  constructor(
+    private bsModalRef: BsModalRef,
+    private highlightService: HighlightService,
+    private alertService: AlertService
+  ) { }
+
+  selectHighlight(selectedHighlight: Highlight) {
+    this.selectHighlightToEdit.emit(selectedHighlight);
+    this.bsModalRef.hide();
+  }
+
+  deleteHighlight(selectedHighlight: Highlight) {
+    this.highlightService.delete(selectedHighlight.id)
+      .subscribe(() => {
+        const alertConfig: AlertConfig = {
+          type: AlertType.success,
+          mainMessage: 'The highlight was succesfully deleted',
+          dismissible: true,
+          timeout: this.alertService.defaultTimeout
+        };
+
+        this.alertService.pushAlert(alertConfig);
+        this.bsModalRef.hide();
+      });
+  }
+
+  addHighlight() {
+    this.selectAddHighlight.emit();
+    this.bsModalRef.hide();
+  }
```

Comment:
The code looks good! One thing I noted, I think (almost) every method in the typescript files has some sort of documentation. I suggest doing the same 😄 . For the rest: Good job